app/assets/javascripts/discourse/lib/utilities.js

```diff
@@ -251,7 +251,8 @@ Discourse.Utilities = {
 
   uploadLocation: function(url) {
     if (Discourse.CDN) {
-      return Discourse.CDN.startsWith('//') ? "http:" + Discourse.getURLWithCDN(url) : Discourse.getURLWithCDN(url);
+      url = Discourse.getURLWithCDN(url);
+      return url.startsWith('//') ? 'http:' + url : url;
```

Comment:
I was going to ask about that :)

On Mon, Jun 20, 2016 at 2:35 PM, Régis Hanol <notifications@github.com>
wrote:

> In app/assets/javascripts/discourse/lib/utilities.js
> <https://github.com/discourse/discourse/pull/4279#discussion_r67771120>:
>
> > @@ -251,7 +251,8 @@ Discourse.Utilities = {
> >
> >    uploadLocation: function(url) {
> >      if (Discourse.CDN) {
> > -      return Discourse.CDN.startsWith('//') ? "http:" + Discourse.getURLWithCDN(url) : Discourse.getURLWithCDN(url);
> > +      url = Discourse.getURLWithCDN(url);
> > +      return url.startsWith('//') ? 'http:' + url : url;
>
> Is it always "http"? Might want to check use_https site setting
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/discourse/discourse/pull/4279/files/9bf381b95c5f29fce9c7a40e7a3d55da2ba8d9df#r67771120>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AC1orfvSEpb_OpWzMsjxHC-vjcdBXMIsks5qNweygaJpZM4I6JxX>
> .
>